docs/lang/articles/basic/syntax.md

```diff
@@ -4,280 +4,484 @@ sidebar_position: 1
+A kernel is the entry point from which Taichi's runtime takes control and the smallest unit for runtime execution. You can define multiple kernels in your program, and each kernel is an *independent* child program. You call a kernel the same way you call a Python function, and you are allowed to switch back and forth between Taichi's runtime and Python's virtual machine.
 
-It is native Python code and will not be compiled by the Taichi compiler.
+Taichi's runtime compiles and executes kernels in the order you call them. It stores compiled kernels in the cache so that the next call to the same kernel does not need to be compiled again. 
 
-:::note
-For people from CUDA, Python-scope = **host** side.
+:::caution WARNING
+You must *not* call a kernel from inside another kernel or from inside a Taichi function. You can only call a kernel directly or from inside a native Python function. To put it differently, you can only call a kernel from the *Python scope*. 
 :::
 
-## Kernels
-
-A Python function decorated by `@ti.kernel` is a **Taichi kernel**:
-
-```python {1}
-@ti.kernel
-def my_kernel():
-    ...
+> A kernel corresponds to the `main()` function in C/C++, or the `__global__` function in CUDA. 
 
-my_kernel()
-```
+### Arguments
 
-Kernels should be called from **Python-scope**. Nested kernels are not supported.
+A kernel can take multiple arguments, supporting scalar, `ti.Matrix`, and `ti.Vector` as argument types. This makes it easier and more flexible to pass values from the Python scope to the Taichi scope. 
 
-:::note
-For people from CUDA, Taichi kernels are similar to `__global__` functions.
+:::caution WARNING
+Arguments in scalar, `ti.Matrix`, or `ti.Vector` are passed by value, so changes to the arguments of a kernel do not affect the original variables in the caller function. 
 :::
 
-### Arguments
-
-Kernels can have multiple arguments, which support passing values from Python-scope to Taichi-scope conveniently.
+Follow these rules when defining arguments:  
 
-:::note
-:::
+- Type hint kernel arguments.
+- Ensure that the total number of elements in the kernel arguments does not exceed a certain upper limit (see below).
 
-Kernel arguments must be type hinted:
+#### Type hint kernel arguments
 
-```python {2}
+```Python
 @ti.kernel
+
 def my_kernel(x: ti.i32, y: ti.f32):
+
     print(x + y)
 
-my_kernel(24, 3.2)  # prints: 27.2
+my_kernel(24, 3.2)  # The system prints 27.2
 ```
 
-:::note
-Taichi supports scalars, `ti.Matrix` and`ti.Vector` as kernel arguments.
-The total number of elements in kernel arguments must not exceed 8 on OpenGL and CC backends, or 64 on other backends.
-The number of elements in a scalar argument is 1, and the number of elements in a `ti.Matrix` or`ti.Vector` is the number of elements inside it.
+#### Ensure that the total number of elements in the kernel arguments does not exceed a certain upper limit
+
+The upper limit for element numbers is backend-specific: 
+
+- 8 on OpenGL or CC backend. 
```

Comment:
Let's remove the mentioning of CC backend for now, because no one is maintaining it